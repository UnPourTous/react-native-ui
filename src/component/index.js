export { default as Button } from './button/Button'
export { default as FlatButton } from './button/FlatButton'
