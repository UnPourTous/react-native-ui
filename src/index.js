import React, { Component } from 'react'
import {
  View,
  TouchableOpacity,
  Text
} from 'react-native-web'

export default class Button extends Component {
  // static propsType = {
  //   title: PropTypes.string.isRequired,
  //   backgroundColor: PropTypes.string
  // }

  // static defaultProps = {
  //   backgroundColor: '#20a0ff'
  // }

  render () {
    const {
      title,
      backgroundColor = '#20a0ff',
      onPress
    } = this.props
    return (
      <TouchableOpacity
        onPress={onPress}>
        <View
          style={{
            flex: 1,
            height: 50,
            backgroundColor,
            alignItems: 'center',
            justifyContent: 'center'
          }}>
          <Text>{title}</Text>
        </View>
      </TouchableOpacity>
    )
  }
}
