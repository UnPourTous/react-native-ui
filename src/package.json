{
  "name": "@unpourtous/react-native-ui",
  "version": "1.0.0",
  "description": "react-native-ui",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "ercihua23",
  "license": "MIT",
  "dependencies": {
    "lodash.merge": "^4.6.0"
  }
}
