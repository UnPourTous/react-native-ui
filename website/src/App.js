/*
 * @flow
 */

import React from 'react'

import AppFrame from './AppFrame'
// import HomePage from './HomePage';
import PageWithSidebar from './PageWithSidebar'
import MDPage from './MDPage'

import {
  TabRouter,
  addNavigationHelpers,
  createNavigator
} from 'react-navigation'

import type { NavigationScreenComponent } from 'react-navigation'

type ScreenOptions = {
  linkName: string,
  icon: string,
  title: string,
};

const NavView = ({ navigation, router }: { navigation: any, router: any }) => {
  const { state } = navigation
  const Component = router.getComponentForState(state)
  return (
    <Component
      navigation={addNavigationHelpers({
        ...navigation,
        state: state.routes[state.index]
      })}
    />
  )
}

type DocPageConfig = {
  doc: string,
  title: string,
  linkName: string,
};

const createDocPage = (
  config: DocPageConfig
): (() => NavigationScreenComponent<*, ScreenOptions>) => {
  const Page: NavigationScreenComponent<*, ScreenOptions> = ({
    navigation
  }: {
    navigation: any,
  }) => <MDPage docPath={config.doc} navigation={navigation} />
  Page.navigationOptions = {
    doc: config.doc,
    title: config.title,
    linkName: config.linkName
  }
  return Page
}

const GuideDocs = createNavigator(
  TabRouter({
    GettingStarted: {
      screen: createDocPage({
        doc: 'guides/Guide-Intro',
        title: 'Hello React-Native-UI', // browser title
        linkName: 'Hello React-Native-UI'
      }),
      path: ''
    }
  })
)(NavView)

GuideDocs.navigationOptions = {
  linkName: 'Getting Started',
  icon: 'pt-icon-flows'
}

const ButtonDocs = createNavigator(
  TabRouter({
    Button: {
      screen: createDocPage({
        doc: 'api/button/Button1',
        title: 'Button1', // browser title
        linkName: 'Button1'
      }),
      path: 'button1'
    },
    Button2: {
      screen: createDocPage({
        doc: 'api/button/Button2',
        title: 'Button2', // browser title
        linkName: 'Button2'
      }),
      path: 'button2'
    }
  })
)(NavView)

ButtonDocs.navigationOptions = {
  linkName: 'Button',
  icon: 'pt-icon-flows'
}

const PopupDocs = createNavigator(
  TabRouter({
    Dialog: {
      screen: createDocPage({
        doc: 'api/popup/Dialog',
        title: 'Button', // browser title
        linkName: 'Button'
      }),
      path: 'button'
    },
    Toast: {
      screen: createDocPage({
        doc: 'api/popup/Toast',
        title: 'Toast', // browser title
        linkName: 'Toast'
      }),
      path: 'toast'
    },
    ActionSheet: {
      screen: createDocPage({
        doc: 'api/popup/ActionSheet',
        title: 'ActionSheet', // browser title
        linkName: 'ActionSheet'
      }),
      path: 'actions-sheet'
    },
    Custom: {
      screen: createDocPage({
        doc: 'api/popup/Custom',
        title: 'Custom', // browser title
        linkName: 'Custom'
      }),
      path: 'custom'
    },
    Animation: {
      screen: createDocPage({
        doc: 'api/popup/Animation',
        title: 'Animation', // browser title
        linkName: 'Animation'
      }),
      path: 'animation'
    }
  })
)(NavView)

PopupDocs.navigationOptions = {
  linkName: 'Popup',
  icon: 'pt-icon-briefcase'
}

const DocsPage = createNavigator(
  TabRouter({
    GuideDocs: {
      screen: GuideDocs,
      path: 'intro'
    },
    ButtonTab: {
      screen: ButtonDocs,
      path: 'button'
    },
    PopupTab: {
      screen: PopupDocs,
      path: 'popup'
    }
  })
)(PageWithSidebar)
DocsPage.navigationOptions = ({
  navigationOptions
}: {
  navigationOptions: any,
}) => ({
  title: `${navigationOptions.title} | React Navigation`
})

const NotFoundError = () =>
  <div className='errorScreen'>
    <h1>Page not found</h1>
  </div>

const App = createNavigator(
  TabRouter({
    // Home: {
    //   screen: HomePage,
    //   path: '',
    // },
    Docs: {
      screen: DocsPage,
      path: 'docs'
    },
    // Blog: {
    //   screen: BlogPage,
    //   path: 'blog',
    // },
    NotFound: {
      screen: NotFoundError,
      navigationOptions: {
        title: 'Page Not Found | React Navigation'
      }
    }
  })
)(AppFrame)

export default App
