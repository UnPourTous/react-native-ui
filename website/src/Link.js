import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavigationActions } from 'react-navigation';

const isModifiedEvent = event =>
  !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);

const Linkable = Inner => {
  class LinkableWrapped extends Component {
    render() {
      return (
        <Inner href={this.getURL()} onClick={this.onClick} {...this.props} />
      );
    }
    getAction = () => {
      const { to, href } = this.props;
      if (typeof to === 'string') {
        return NavigationActions.navigate({ routeName: to });
      } else if (typeof to === 'object' && typeof to.type === 'string') {
        return to;
      } else if (href) {
        const match = href.match(/^\/(.*)/);
        if (match) {
          const pathParts = match[1].split('#');
          const path = pathParts[0];
          let params = {};
          if (pathParts.length) {
            params.hash = pathParts[1];
          }
          const action = this.context.getActionForPathAndParams(path, params);
          return action;
        }
        return null;
      }
    };
    onClick = e => {
      const action = this.getAction();
      if (!isModifiedEvent(e) && action) {
        this.context.dispatch(action);
        e.preventDefault();
      }
    };
    getURL() {
      const action = this.getAction();
      if (!action) return '#';
      return this.context.getURIForAction(action);
    }
    static contextTypes = {
      dispatch: PropTypes.func.isRequired,
      getURIForAction: PropTypes.func.isRequired,
      getActionForPathAndParams: PropTypes.func.isRequired,
    };
  }
  return LinkableWrapped;
};

const Link = Linkable(props => <a {...props} />);

Link.Linkable = Linkable;

export default Link;
