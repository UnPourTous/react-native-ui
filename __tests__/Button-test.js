import {View} from 'react-native';
import React from 'react';
import {Button} from '../src';
import renderer from 'react-test-renderer';
test('renders correctly', () => {
  const tree = renderer.create(
    <Button 
      onPress={() => {
        console.log('onPress')
      }}
      title={'Test Button'} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
