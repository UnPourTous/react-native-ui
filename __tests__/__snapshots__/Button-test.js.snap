// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 2,
          "height": 50,
          "justifyContent": "center",
          "paddingLeft": 50,
          "paddingRight": 50,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;
